meiga/result.py

```diff
@@ -96,32 +87,30 @@ def unwrap(self) -> Union[TS, None]:
             return None
         return cast(TS, self.value)
 
-    def unwrap_or(self, failure_value: Any) -> Union[TS, Any]:
+    def unwrap_or(self, failure_value: TEF) -> Union[TS, TEF]:
         if not self._is_success:
-            return failure_value
+            return cast(TEF, failure_value)
```

Comment:
Is the cast necessary here?
It is for `self._value_failure`, which can be `NoGivenValue` unless asserted otherwise - but here the type is provided in parameters.